src/resolve/resolved-dns-cache.c

```diff
@@ -1265,6 +1268,11 @@ int dns_cache_export_shared_to_packet(DnsCache *cache, DnsPacket *p) {
                         if (!j->shared_owner)
                                 continue;
 
+                        /* RFC6762 7.1: Don't append records with less than
+                         * half the TTL remaining as known answers. */
+                        if ((t >= j->until) || (j->until - t) < (j->rr->ttl * (USEC_PER_SEC / 2)))
```

Comment:
Please drop redundant `()`, and use `usec_sub_unsigned()`:
`if (usec_sub_unsigned(t, j->until) < j->rr->ttl * USEC_PER_SEC / 2)`